plugins/ccgx/fu-ccgx-dock-bb.c

```diff
@@ -0,0 +1,734 @@
+/*
+ * Copyright (C) 2020 Cypress Semiconductor Corporation.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fwupd-error.h"
+
+#include "fu-common-version.h"
+#include "fu-usb-device.h"
+
+#include "fu-ccgx-dock-bb.h"
+#include "fu-ccgx-common.h"
+#include "fu-ccgx-hpi.h"
+#include "fu-ccgx-hid.h"
+#include "fu-ccgx-cyacd-firmware.h"
+#include "fu-ccgx-cyacd-file.h"
+
+/* i2c slave address for pd device */
+#define PD_I2C_SLAVE_ADDRESS		0x08
+
+/* hid Interface number */
+#define USB_HID_INF_NUM			1
+
+/* usb i2c interface number */
+#define USB_I2C_INF_NUM			0
+
+/* gen2 dock model name */
+#define CCGX_GEN2_DOCK_MODEL_NAME	"Gen2"
+
+struct _FuCcgxDockBb {
+	FuUsbDevice	 parent_instance;
+	guint16		 usb_inf_num;			/* usb Interface Number */
+	CyHPIHandle	*hpi_handle;			/* hpi handle for pd device i2c */
+	PDDeviceData	 pd_device_data;		/* pd device Information data */
+	DMDevice	 dm_device;			/* device manager device type */
+	guint16		 quirks_silicon_id;		/* silicon id in quirks */
+	guint16		 quirks_fw_app_type;		/* fw application type in quirks */
+	gboolean	 flag_dm_has_child;		/* flag that indicate device manager has child */
+	FWImageType	 fw_image_type;			/* firmware Image type */
+	gboolean	 fw_primary_update_only;	/* only update primary image */
+	gchar		*fw_update_message;		/* update message */
+	gchar		*fw_update_message_primary;	/* update message postfix for primary */
+	gchar		*fw_update_message_backup;	/* update message postfix for backup */
+	gchar		*model_name;			/* dock model name */
+	gboolean	 fw_update_success;		/* fw update success flag */
```

Comment:
do you actually need this flag?  Can't you use use the `fu_device_get_update_state`/`fu_device_set_update_state` functions?

Reply:
I can use fu_device_get_update_state instead of fw_update_success flag.
Do I need to set update status using fu_device_set_update_state or it is set automatically?

Reply:
I fixed it. remove fw_update_success and use fu_device_get_update_state.